app/models/registry.rb

```diff
@@ -0,0 +1,61 @@
+# model that represents a registry
+class Registry < ActiveRecord::Base
+  has_many :registry_mirrors, dependent: :destroy
+  has_one :certificate_service, as: :service, dependent: :destroy
+  has_one :certificate, through: :certificate_service
+
+  validates :name, presence: true, uniqueness: true
+  validates :url, presence: true, uniqueness: true, url: { schemes: ["https", "http"] }
+
+  class << self
+    # create or update suse Registry model
+    # rubocop:disable Metrics/CyclomaticComplexity,Metrics/PerceivedComplexity
+    def configure_suse_registry(suse_registry_mirror_params)
+      if suse_registry_mirror_params.blank?
+        cleanup
+        return []
+      end
+
+      errors      = []
+      cert        = suse_registry_mirror_params["certificate"]
+      mirror      = suse_registry_mirror_params["mirror"]
+      name        = suse_registry_mirror_params["name"] || "suse"
```

Comment:
the codeclimate issue is caused by this line here. i suggest we ignore it (as i did with rubocop, which reported the same thing), as there is no better way to do it i think. can we configure codeclimate somehow to let that pass?

and the other Q: do we even need codeclimate when rubocop is testing the same thing?